src/modes/OptimInterface.jl

```diff
@@ -206,13 +206,13 @@ function _optimize(
     args...; 
     kwargs...
 )
-    return _optimize(model, f, f.vi[DynamicPPL.SampleFromPrior()], Optim.LBFGS(), args...; kwargs...)
+    return _optimize(model, f, f.varinfo[f.sampler], Optim.LBFGS(), args...; kwargs...)
 end
 
 function _optimize(
     model::Model, 
     f::OptimLogDensity, 
-    init_vals::AbstractArray = f.vi[DynamicPPL.SampleFromPrior()], 
+    init_vals::AbstractArray = f.varinfo[f.sampler],
```

Comment:
L226 needs to be changed in a similar way to this line

Reply:
Thanks, I fixed it!